extensions/wikia/UserLogin/js/FacebookLogin.js

```diff
@@ -121,10 +124,19 @@
 				// begin ajax call performance tracking
 				this.bucky.timer.start('loginCallbackAjax');
 
+				pageUrl = new QueryString();
+				returnToURL = pageUrl.getVal('returnto');
+				if (returnToURL) {
+					returnToQuery = pageUrl.getVal('returntoquery');
+				} else {
+					returnToURL = encodeURIComponent(window.wgPageName);
+					returnToQuery = encodeURIComponent(window.location.search.substring(1));
```

Comment:
Just putting this out there - you could use `pageUrl.getVals()` to get a key/value array of the query string params. It's up to you if you think that would make this code cleaner.